docs/DeveloperGuide.md

```diff
@@ -1,34 +1,972 @@
+2. Then, `removeTakenByPrerequisites` checks the list of prerequisites of `newModule` and 
+removes the ones that have been marked as taken.
+3. Finally, `add` calls `ArrayList.add()` to add `newModule` to the list.
+
+##### Deleting a module
+
+For deleting a module to the list, the methods defined are `delete` and `removeFromPreRequisiteModuleRequiredBy`. To
+delete an existing module from the list, say `existingModule`, `delete` is called to perform 2 steps:
+1. Firstly, `delete` calls `ArrayList.remove()` to delete `existingModule` from the list.
+2. Then, `removeFromPreRequisiteModuleRequiredBy` is called to remove `existingModule` from its pre-requisite modules' 
+   requiredBy list.
+   
+##### Mark module as taken
+
+For marking a module as taken, the methods defined are `markAsTaken` and `removeFromModuleUntakenPrerequisites`. To mark
+an existing module as taken, say `existingModule`, `markAsTaken` is called to perform 2 steps:
+1. First, `markAsTaken` calls `Module.setStatus` and sets the status of `existingModule` to "taken".
+2. Then, `removeFromModuleUntakenPrerequisites` is called to remove `existingModule` from the prerequisitesUntaken table
+from the list of modules that require `existingModule` as a prerequisite.
+
+Given below is the class diagram of `ModuleList` showing its 3 main functions.
+
+![archi](./images/ModuleListClassDiagram.png)
+
+<sup>***Figure 3.4.2.1** UML class diagram for ModuleList class*</sup>
+
+Apart from these 3 operations, the `ModuleList` class also defines getter and setter methods to retrieve values such
+as the entire list or an individual module from the list according to different parameters such as module code or index. 
+
+### 3.5 Storage Component
+
+Class Diagram:
+
+![archi](./images/storageClassDiagram.jpg)
+
+<sup>***Figure 3.5.1** UML class diagram for Storage package*</sup>
+
+The `Storage` Component, 
+- Can save `module` objects in the `moduleList` in a JSON format and read them back
+
+![archi](./images/storageObjectDiagram.jpg)
+
+<sup>***Figure 3.5.2** UML object diagram for an instance of storage object*</sup>
+
+### 3.6 Common classes
+The common class used by multiple components in this app are in the `exception` package. The `exceptions` are thrown
+when an error occurs. The method catches the exceptions and prints out the respective error message. 
+
+Each `exception` is specified by the name and description.
+
+## 4. Implementation
+
+This section elaborates on some details about how certain features are implemented.
+
+### 4.1 UI
+
+UI component
+* Executes user command using Logic Component
+* Listens to calls from Model data so that UI can print the results.
+
+
+### 4.2 Parser
+
+The parser feature has 3 primary responsibilities: 
+1. Identify the command the user wants to run.
+2. Extract the relevant parameters and flags required to run the command
+3. Create a new `Command` object and hand it over to `iGraduate` to execute
+
+![archi](./images/ParserSequenceDiagram.png)
```

Comment:
Are all the self call methods necessary to be included?

Reply:
Agreed, seems like it could be excluded for more readability.